src/OrchardCore.Modules/OrchardCore.AuditTrail/Services/AuditTrailManager.cs

```diff
@@ -0,0 +1,239 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+using OrchardCore.AuditTrail.Extensions;
+using OrchardCore.AuditTrail.Indexes;
+using OrchardCore.AuditTrail.Models;
+using OrchardCore.AuditTrail.Providers;
+using OrchardCore.AuditTrail.Services.Models;
+using OrchardCore.AuditTrail.Settings;
+using OrchardCore.Entities;
+using OrchardCore.Modules;
+using OrchardCore.Settings;
+using YesSql;
+using IYesSqlSession = YesSql.ISession;
+
+namespace OrchardCore.AuditTrail.Services
+{
+    public class AuditTrailManager : IAuditTrailManager
+    {
+        private readonly IClock _clock;
+        private readonly IStringLocalizer T;
+        private readonly IYesSqlSession _session;
+        private readonly IHttpContextAccessor _hca;
+        private readonly ISiteService _siteService;
+        private readonly Entities.IIdGenerator _iidGenerator;
+        private readonly IEnumerable<IAuditTrailEventHandler> _auditTrailEventHandlers;
+        private readonly IEnumerable<IAuditTrailEventProvider> _auditTrailEventProviders;
+
+        public ILogger Logger { get; set; }
+
+
+        public AuditTrailManager(
+            IClock clock,
+            IYesSqlSession session,
+            ISiteService siteService,
+            IHttpContextAccessor hca,
+            ILogger<AuditTrailManager> logger,
+            Entities.IIdGenerator iidGenerator,
```

Comment:
Suggest following the pattern used everywhere else.
`IAuditTrailEventIdGenerator` and the implementation injects `IIdGenerator`